tests/model_selection/test_hyperband.py

```diff
@@ -0,0 +1,298 @@
+import numpy as np
+import scipy.stats
+from sklearn.linear_model import SGDClassifier
+
+import dask
+import dask.array as da
+from dask.distributed import Client, wait
+from distributed.utils_test import loop, cluster, gen_cluster  # noqa: F401
+from tornado import gen
+
+from dask_ml.datasets import make_classification
+from dask_ml.model_selection import HyperbandCV
+from dask_ml.wrappers import Incremental
+from dask_ml.model_selection._hyperband import _partial_fit
+from dask_ml.utils import ConstantFunction
+from distributed.metrics import time
+
+import pytest
+
+
+@pytest.mark.parametrize(
+    "array_type,library",  # noqa: F811
+    [("dask.array", "dask-ml"), ("numpy", "sklearn"), ("numpy", "test")],
+)
+def test_sklearn(array_type, library, loop, max_iter=27):
+    with cluster() as (s, [a, b]):
+        with Client(s["address"], loop=loop):
+            n, d = (1000, 4)
+            chunk_size = n // 2
+            X, y = make_classification(
+                n_samples=n, n_features=d, random_state=42, chunks=chunk_size
+            )
+            if array_type == "numpy":
+                X = X.compute()
+                y = y.compute()
+                chunk_size = X.shape[0]
+
+            sgd_params = {"alpha": np.logspace(-3, 0, num=1000)}
+            kwargs = dict(tol=-np.inf, penalty="elasticnet", random_state=42)
+            if library == "sklearn":
+                model = SGDClassifier(**kwargs)
+                params = sgd_params
+            elif library == "dask-ml":
+                model = Incremental(SGDClassifier(**kwargs))
+                params = sgd_params
+            elif library == "test":
+                model = ConstantFunction()
+                params = {"value": np.linspace(0, 1, num=1000)}
+            else:
+                raise ValueError
+
+            search = HyperbandCV(
+                model,
+                params,
+                max_iter=max_iter,
+                random_state=42,
+                asynchronous=False,
+            )
+            search.fit(X, y, classes=da.unique(y))
+
+            score = search.best_estimator_.score(X, y)
+            if library == "sklearn":
+                assert score > 0.7
+            if library == "dask-ml":
+                assert score > 0.5
```

Comment:
Why is the dask-ml score different than the sklearn score here?

Reply:
Because of the chunking. We split the array into two pieces, so the dask-ml model sees half as many examples.

What should I change?

Reply:
Hrm, I didn't realize that results would depend on the chunking of the dataset.

Reply:
This happens because we call `partial_fit` over each chunk of the dataset, and the input `max_iter` controls the maximum number of calls to `partial_fit`. I'll add a note about this to HyperbandCV's docstring.

Reply:
If you increase `max_iter` here will it converge to the scikit-learn result?